Sources/SwiftParser/Patterns.swift

```diff
@@ -217,9 +217,20 @@ extension Parser {
       while !self.at(.eof, .rightParen) && keepGoing && loopProgress.evaluate(currentToken) {
         // If the tuple element has a label, parse it.
         let labelAndColon = self.consume(if: .identifier, followedBy: .colon)
-        let (label, colon) = (labelAndColon?.0, labelAndColon?.1)
+        var (label, colon) = (labelAndColon?.0, labelAndColon?.1)
+
+        if colon == nil, peek().rawTokenKind == .identifier, peek().tokenText.isStartingWithSuppercase {
+          label = consumeAnyToken()
```

Comment:
What happens if `label != nil` here. Would we be dropping the existing label in that case?

Reply:
Yes it will. Fixed it